foundationdb/src/tuple/mod.rs

```diff
@@ -224,11 +278,115 @@ mod tests {
 
     #[test]
     fn test_eq() {
+        assert_eq!("string".to_vec(), "string".to_string().to_vec());
+
+        assert_eq!(
+            ("string", "string".to_string()).to_vec(),
+            ("string".to_string(), "string").to_vec()
+        );
+    }
+
+    #[test]
+    fn test_encode_recursive_tuple() {
+        assert_eq!(
+            &("one", ("two", 42)).to_vec(),
+            &[2, 111, 110, 101, 0, 5, 2, 116, 119, 111, 0, 21, 42, 0]
+        );
         assert_eq!(
-            "string".to_vec(),
-            "string".to_string().to_vec()
+            &("one", ("two", 42, ("three", 33))).to_vec(),
+            &[
+                2, 111, 110, 101, 0, 5, 2, 116, 119, 111, 0, 21, 42, 5, 2, 116, 104, 114, 101, 101,
+                0, 21, 33, 0, 0,
+            ]
         );
 
-        assert_eq!("string".to_vec(), ("string",).to_vec());
```

Comment:
Isn't this test still relevant, and even important?

Reply:
I added a separate set of tests for nested tuples. I can add this back.

Reply:
Yes I noticed, but I think this one is unique in that I haven't seen any other test comparing a single element to a tuple with a single element.

Reply:
technically there is this one :`assert_eq!(&(1,).to_vec(), &1.to_vec());` but it's ok, I've added this back.